mypy/dmypy_server.py

```diff
@@ -152,6 +153,7 @@ def serve(self) -> None:
                         conn, addr = sock.accept()
                     except socket.timeout:
                         print("Exiting due to inactivity.")
+                        TypeState.reset_all_subtype_caches()
```

Comment:
Maybe move this call to a method (say, `free_global_state`) since this call and the identical one below must stay in sync, and by having them in a single method makes that easier.